notebooks/Open Bandit Pipeline Walkthrough.ipynb

```diff
@@ -0,0 +1,396 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Demo of Open Bandit Pipeline\n",
+    "\n",
+    "![](https://github.com/st-tech/zr-obp/blob/master/images/overview.png?raw=true)\n",
+    "\n",
+    "**[Official documentation for OBP](https://zr-obp.readthedocs.io/en/latest/)**\n",
+    "\n",
+    "## Quick Reference:\n",
+    "\n",
+    "**OBP: Open Bandit Pipeline** - this software library\n",
+    "\n",
+    "**OBD: Open Bandit Dataset** - the dataset supplied with it\n",
+    "\n",
+    "**OPE: off-policy evaluation** - the process of determining how a policy _other than the one that was really run_ woudl have performed"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Dataset Loader\n",
+    "\n",
+    "The first part of the Open Bandit Pipeline (OBP) is the dataset loader. For the Open Bandit Dataset (OBD), the loader is `opb.dataset.OpenBanditDataset` ([docs](https://zr-obp.readthedocs.io/en/latest/_autosummary/obp.dataset.real.html#obp.dataset.real.OpenBanditDataset)). \n",
+    "\n",
+    "As with many classes in the OBP, the dataset modules are implemented with [dataclasses](https://docs.python.org/3.7/library/dataclasses.html).\n",
+    "\n",
+    "The dataset module inherits from `obp.dataset.base.BaseRealBanditDatset` ([docs](https://zr-obp.readthedocs.io/en/latest/_autosummary/obp.dataset.base.html#module-obp.dataset.base)) and should implement three methods:\n",
+    "- `load_raw_data()`: Load an on-disk representation of the dataset into the module. Used during initialization.\n",
+    "- `pre_process()`: Perform any preprocessing needed to transform the raw data representation into a final representation.\n",
+    "- `obtain_batch_bandit_feedback()`: Return a dictionary containing (at least) keys: `[\"action\",\"position\",\"reward\",\"pscore\",\"context\"]`\n",
+    "\n",
+    "It is also helpful if the dataset module exposes a property `len_list`, which is how many items the bandit shows the user at a time. Often the answer is 1, though in the case of OBD it's 3."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 62,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/Users/danturkel/.pyenv/versions/3.7.9/envs/sd_bandits/lib/python3.7/site-packages/numpy/lib/arraysetops.py:580: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison\n",
```

Comment:
This warning is unclear to me but I assume we can disregard.

Reply:
looks like something OBP will need to fix, not us. luckily it's just a futurewarning.